src/megameklab/com/ui/StartupGUI.java

```diff
@@ -0,0 +1,421 @@
+        add(panTitle, c);
+        // Right Column
+        c.insets = new Insets(2, 2, 2, 10);
+        c.fill = GridBagConstraints.BOTH;
+        c.weightx = 1.0; c.weighty = 1.0;
+        c.ipadx = 0; c.ipady = 0;
+        c.gridheight = 1;
+        c.gridx = 1; c.gridy = 0;
+        add(labVersion, c);
+        c.gridy++;
+        add(btnLoadUnit, c);
+        c.gridy++;
+        add(btnNewMek, c);
+        c.gridy++;
+        add(btnNewVee, c);
+        c.gridy++;
+        add(btnNewSupportVee, c);
+        c.gridy++;
+        add(btnNewProto, c);
+        c.gridy++;
+        add(btnNewBA, c);
+        c.gridy++;
+        add(btnNewPbi, c);
+        c.gridy++;
+        add(btnNewAero, c);
+        c.gridy++;
+        add(btnNewDropper, c);
+        c.gridy++;
+        add(btnNewLargeCraft, c);
+        c.gridy++;
+        add(btnQuit, c);
+        
+        frame.setResizable(false);
+        frame.getContentPane().setLayout(new BorderLayout());
+        frame.getContentPane().add(this, BorderLayout.CENTER);
+        frame.addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(WindowEvent e) {
+                frame.setVisible(false);
+                System.exit(0);
+            }
+        });
+        frame.validate();
+        frame.pack();
+        
+        // Determine the location of the window
+        int w = frame.getSize().width;
+        int h = frame.getSize().height;
+        int x = (currentMonitor.getWidth()-w)/2;
+        int y = (currentMonitor.getHeight()-h)/2;
+        frame.setLocation(x, y);
+        
+        frame.setVisible(true);
+    }
+    
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+        if (backgroundIcon == null){
+            return;
+        }
+        int w = getWidth();
+        int h = getHeight();
+        int iW = backgroundIcon.getWidth();
+        int iH = backgroundIcon.getHeight();
+        // If the image isn't loaded, prevent an infinite loop
+        if ((iW < 1) || (iH < 1)) {
+            return;
+        }
+        for (int x = 0; x < w; x+=iW){
+            for (int y = 0; y < h; y+=iH){
+                g.drawImage(backgroundIcon, x, y,null);
+            }
+        }
+     }
+
+    /**
+     * This function will create a new mainUI frame (via the loading dialog) for the 
+     * given unit type and get rid of the splash screen
+     * @param type an <code>int</code> corresponding to the unit type to construct
+     */
+    private void newUnit(long type) {
+        newUnit(type, false, false, null);
+    }
+    
+    private void newUnit(long type, boolean primitive, boolean industrial, Entity en) {
+        frame.setVisible(false);
+        LoadingDialog ld = new LoadingDialog(frame, type, primitive, industrial, en);
+        ld.setVisible(true);
+    }
+    
+    private void loadUnit() {
+        UnitLoadingDialog unitLoadingDialog = new UnitLoadingDialog(frame);
```

Comment:
This is the place where the first time the button is pushed it shows the UnitLoadingDialog and then disappears with nothing else happening.

Reply:
Duh, check the log stupid. Here is where the problem seems to be:

```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at megamek.client.ui.swing.AdvancedSearchDialog$WeaponsTableModel.getColumnClass(AdvancedSearchDialog.java:1118)
	at javax.swing.table.TableRowSorter.useToString(TableRowSorter.java:241)
	at javax.swing.DefaultRowSorter.updateUseToString(DefaultRowSorter.java:624)
	at javax.swing.DefaultRowSorter.sort(DefaultRowSorter.java:573)
	at javax.swing.DefaultRowSorter.allRowsChanged(DefaultRowSorter.java:852)
	at javax.swing.JTable.notifySorter(JTable.java:4264)
	at javax.swing.JTable.sortedTableChanged(JTable.java:4124)
	at javax.swing.JTable.tableChanged(JTable.java:4401)
	at javax.swing.table.AbstractTableModel.fireTableChanged(AbstractTableModel.java:296)
	at javax.swing.table.AbstractTableModel.fireTableDataChanged(AbstractTableModel.java:198)
	at megamek.client.ui.swing.AdvancedSearchDialog$WeaponsTableModel.setData(AdvancedSearchDialog.java:1138)
	at megamek.client.ui.swing.AdvancedSearchDialog.populateWeaponsAndEquipmentChoices(AdvancedSearchDialog.java:1010)
	at megamek.client.ui.swing.AdvancedSearchDialog.<init>(AdvancedSearchDialog.java:378)
	at megamek.client.ui.swing.UnitSelectorDialog.<init>(UnitSelectorDialog.java:197)
	at megameklab.com.ui.StartupGUI.loadUnit(StartupGUI.java:379)
```

Reply:
Need to initialized EquipmentTypes. Fixed in commit 40f38aca.